tst/EnergyPlus/unit/ScheduleManager.unit.cc

```diff
@@ -1451,3 +1456,110 @@ TEST_F(EnergyPlusFixture, ScheduleFileDSTtoggleOptionTest)
     state->dataEnvrn->DSTIndicator = 0; // Tells the simulation that we're NOT currently observing daylight savings
     EXPECT_DOUBLE_EQ(ScheduleManager::LookUpScheduleValue(*state, sch4idx, state->dataGlobal->HourOfDay, state->dataGlobal->TimeStep), 1.0);
 }
+
+TEST_F(EnergyPlusFixture, ShadowCalculation_CSV_extra_parenthesis)
+{
+
+    // 9753 - Test backward compat:
+    // a CSV exported with the extra '()' at the end (22.2.0 and below) should still be importable in E+ without crashing
+    const fs::path scheduleFile = configured_source_directory() / "tst/EnergyPlus/unit/Resources/shading_data_2220.csv";
+
+    std::string const idf_objects = delimited_string({
+        "Schedule:File:Shading,",
+        "  " + scheduleFile.string() + ";              !- Name of File",
+    });
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    state->dataGlobal->NumOfTimeStepInHour = 4; // must initialize this to get schedules initialized
+    state->dataGlobal->MinutesPerTimeStep = 15; // must initialize this to get schedules initialized
+    state->dataGlobal->TimeStepZone = 0.25;
+    state->dataGlobal->TimeStepZoneSec = state->dataGlobal->TimeStepZone * DataGlobalConstants::SecInHour;
+    state->dataEnvrn->CurrentYearIsLeapYear = false;
+
+    EXPECT_FALSE(state->dataScheduleMgr->ScheduleFileShadingProcessed);
+    ScheduleManager::ProcessScheduleInput(*state); // read schedules
```

Comment:
Backward compat test. Without backward compat fix, it segfaults on the ProcessScheduleInput

Reply:
:+1: